src/middlewared/middlewared/plugins/kerberos.py

```diff
@@ -0,0 +1,613 @@
+import asyncio
+import base64
+import datetime
+import enum
+import os
+import subprocess
+import time
+from middlewared.schema import accepts, Any, Bool, Cron, Dict, Int, List, Patch, Path, Str
+from middlewared.service import CallError, ConfigService, CRUDService, Service, item_method, private, ValidationErrors
+from middlewared.utils import run, Popen
+
+class keytab(enum.Enum):
+    SYSTEM = '/etc/krb5.keytab'
+    SAMBA = '/var/db/samba4/private/samba.keytab'
+
+class KerberosService(ConfigService):
+    """
+    :start:  - configures kerberos and performs kinit if needed
+    :stop:   - performs kdestroy and clears cached klist output
+    :status: - returns false if there is no kerberos tgt or if it is expired 'klist -t'
+    :renew:  - compares current time with expiration timestamp in tgt. Issues 'kinit -R' if needed.
+               uses cached klist output if available.
+    :update: - this modifies the krb5.conf. There is a rudamentary parser for the auxiliary
+               parameters.
```

Comment:
These need to be documented in their respective methods to be part of the API documentation. In fact we need every public method to be documented.

Reply:
Okay. Done.